samples/snap-geometry-edits-with-utility-network-rules/src/main/java/com/esri/arcgismaps/sample/snapgeometryeditswithutilitynetworkrules/components/SnapGeometryEditsWithUtilityNetworkRulesViewModel.kt

```diff
@@ -0,0 +1,512 @@
+                if (identifiedFeature.featureTable?.geometryType != GeometryType.Point) {
+                    resetSelections()
+                    return@launch
+                } else if (
+                    selectedFeature != null
+                    && identifiedFeature != selectedFeature
+                    && selectedFeature?.featureTable?.layer is FeatureLayer
+                ) {
+                    // If a feature is already selected and the tapped feature is not the selected
+                    // feature then clear the previous selection
+                    (selectedFeature?.featureTable?.layer as FeatureLayer).clearSelection()
+                }
+
+                // Update the selected feature and select it on the layer
+                selectedFeature = identifiedFeature
+                selectedFeature?.let { feature ->
+                    (feature.featureTable?.layer as FeatureLayer).selectFeature(feature)
+
+                    // Create a utility element for the selected feature using the utility network
+                    arcGISMap.utilityNetworks.first().createElementOrNull(feature)?.let { element ->
+                        // Update values for UI based on the selected feature
+                        _assetGroupNameState.value = element.assetGroup.name
+                        _assetTypeNameState.value = element.assetType.name
+                        _isEditButtonEnabled.value = true
+                        setSnapSettings(element.assetType)
+                    } ?: run {
+                        messageDialogVM.showMessageDialog("Error creating UtilityElement")
+                        return@launch
+                    }
+
+                    // Start the editing session once feature is selected.
+                    editFeatureGeometry()
+                }
+            }
+        }
+    }
+
+    /**
+     * Creates SnapRules based on the given asset type, synchronizes the snap sources using these
+     * rules, then updates the snap sources list used by the UI
+     */
+    private suspend fun setSnapSettings(assetType: UtilityAssetType) {
+        // Get the snap rules associated with the asset type
+        val snapRules = SnapRules.create(arcGISMap.utilityNetworks.first(), assetType).getOrThrow()
+
+        geometryEditor.snapSettings.apply {
+            // Synchronize the snap source collection with the map's operational layers using the snap
+            // rules. Setting SnapSourceEnablingBehavior.SetFromRules will enable snapping for the
+            // layers and sublayers specified in the snap rules.
+            syncSourceSettings(snapRules, SnapSourceEnablingBehavior.SetFromRules)
+
+            // Enable snapping for the graphics overlay as this will not be affected by the given
+            // SnapSourceEnablingBehavior.setFromRules
+            sourceSettings.first { sss -> sss.source == graphicsOverlay }.isEnabled = true
```

Comment:
```suggestion
            sourceSettings.first { it.source == graphicsOverlay }.isEnabled = true
```

it might be clearer in this case since sss doesn't mean too much for those who don't know the API